newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js

```diff
@@ -183,30 +180,34 @@ export const HomePage = React.memo<Props>(
         gameDetailsCurrentTab,
         setGameDetailsCurrentTab,
       ] = React.useState<GameDetailsTab>('details');
+      const { routeArguments, removeRouteArguments } = React.useContext(
+        RouterContext
+      );
 
       const windowWidth = useResponsiveWindowWidth();
       const isMobile = windowWidth === 'small';
       const {
         values: { showGetStartedSectionByDefault },
       } = React.useContext(PreferencesContext);
-      const initialTab = showGetStartedSectionByDefault
+      const isShopRequestedAtOpening = React.useRef<boolean>(
+        isShopRequested(routeArguments)
```

Comment:
nice, are we sure the routeArguments are correctly set before this component is loaded?

Reply:
I believe this is safe, routeArguments are read from the URL (or the command line, on Electron) and are immediately available when the page (or app, on Electron) starts (see calls to `Window.getArguments()`).